File: task-3/Haskell/Sub-Task 3/main.hs
import Text.Printf

main :: IO ()
main = do
  putStr "Enter rows: "
  n <- readLn :: IO Int
  let counter = n `mod` 2
      buildRow i = do
        let stars = replicate counter '*'
            spaces = replicate (div (n - counter) 2) ' '
        printf "%s%s%s\n" spaces stars spaces
      nextCounter i counter = if i < div n 2 then counter + 2 else counter - 2
      buildPattern n counter =
        let updatedCounter = if counter == 0 then counter + 2 else counter
        in buildRow 0 updatedCounter : map (\i -> buildRow i (nextCounter i updatedCounter)) [1..n-1]

  mapM_ (flip buildRow) (buildPattern n counter)

File: task-3/C/Sub-Task 3/main.c

#include <stdio.h>
#include <math.h>

void replicate(char ch, int num)
{
	for(int i=0; i<num; ++i)
	{
		printf("%c", ch);
	}
}

int main()
{
	int n;
	printf("Enter rows: ");
	scanf("%d", &n);
	int counter = n%2;

	for(int i =0; i<n; ++i)
	{
		if(counter == 0)
		{
			counter += 2;
			continue;
		}

		int half = floor((n-counter)/2);
		replicate(' ', half);
		replicate('*', counter);
		replicate(' ', half);
		printf("\n");

		if(i < floor(n/2))
		{
			counter += 2;
		} else {
			counter -= 2;
		}
	}

	return 0;
}